metrics/metrics.go

```diff
@@ -0,0 +1,44 @@
+package metrics
+
+import "github.com/prometheus/client_golang/prometheus"
+
+type Observer interface {
+	Observe(val float64, labels ...string)
+
+	// for now we will tightly couple to the prometheus collector type
+	// the go otel metrics sdk also has a prometheus adapter that implements this interface.
+	prometheus.Collector
+}
+
+type Metrics struct {
+	TMIMsgsCount    Observer
+	TMICommandCount Observer
+	LearnedCount    Observer
+	ForgotCount     Observer
+	SpeakLatency    Observer
+	LearnLatency    Observer
+}
+
+func (m Metrics) Collectors() []prometheus.Collector {
+	return []prometheus.Collector{
+		m.ForgotCount,
+		m.LearnedCount,
+		m.SpeakLatency,
+		m.TMICommandCount,
+		m.TMIMsgsCount,
+		m.LearnLatency,
+	}
+}
+
+// for testing purposes.
+type StubMetric struct{}
```

Comment:
If this is for testing, then it should be in a _test.go file. Right now it seems to be unused, though.

Reply:
I originally added this because I figured adding *Metrics as a member on Robot would cause some tests to fail. I removed this since there were no such tests.